test/functional/test_framework/test_node.py

```diff
@@ -243,6 +244,20 @@ def wait_for_rpc_connection(self):
             time.sleep(1.0 / poll_per_s)
         self._raise_assertion_error("Unable to connect to bitcoind")
 
+    def wait_for_cookie_credentials(self):
+        self.log.debug("Waiting for cookie credentials")
+        # Poll at a rate of four times per second
+        poll_per_s = 4
+        for _ in range(poll_per_s * self.rpc_timeout):
+            try:
+                user, passphrase = get_auth_cookie(self.datadir, self.chain)
+                self.log.debug("Cookie credentials retrieved: user={}, passphrase={}".format(user, passphrase))
```

Comment:
nit: the values aren't needed, only their presence, so no need to log them?

Reply:
done